app/controllers/RecordsController.scala

```diff
@@ -2,20 +2,35 @@ package controllers
 
 import java.util.UUID
 
-import auth.OidcSecurity
+import auth.TokenSecurity
 import configuration.{GraphQLConfiguration, KeycloakConfiguration}
 import javax.inject.{Inject, Singleton}
 import org.pac4j.play.scala.SecurityComponents
 import play.api.i18n.I18nSupport
-import play.api.mvc.{Action, AnyContent, Request}
+import play.api.mvc.{Action, AnyContent, Request, RequestHeader}
+import services.ConsignmentService
+
+import scala.concurrent.ExecutionContext
 
 @Singleton
 class RecordsController @Inject()(val controllerComponents: SecurityComponents,
                                   val graphqlConfiguration: GraphQLConfiguration,
-                                  val keycloakConfiguration: KeycloakConfiguration
-                                 ) extends OidcSecurity with I18nSupport {
+                                  val keycloakConfiguration: KeycloakConfiguration,
+                                  consignmentService: ConsignmentService
+                                 )(implicit val ec: ExecutionContext) extends TokenSecurity with I18nSupport {
+
+  private def getRecordProcessingProgress(request: Request[AnyContent], status: Status, consignmentId: UUID)(implicit requestHeader: RequestHeader) = {
```

Comment:
I think that's a good idea, I'm guessing that would then give us the chance to deal with a failure case too since there could then be some sort of case statement/conditional?